### PR TITLE
Apple TV support 1: podspec and code changes for commonly used modules

### PR DESCRIPTION
**Pull request by douglowder**
# Why

First of a series of PRs to add Apple TV support for many (but not all) Expo SDK modules.

This PR makes Apple TV able to compile for apps using `react-native-tvos@0.72.4-0` and ONLY the following modules:

- expo
- expo-application
- expo-constants
- expo-eas-client
- expo-file-system
- expo-font
- expo-json-utils
- expo-keep-awake
- expo-manifests
- expo-modules-core
- expo-splash-screen
- expo-structured-headers
- expo-updates-interface
- expo-updates

# How

- Add tvOS to podspecs
- Native code changes wrapped in `#if TARGET_OS_TV` (for Objective C) or `#if os(tvOS)` (for Swift)
- Import source for the `ASN1Decoder` Swift framework used by `expo-updates`, as the public podspec for that framework does not yet support tvOS

# Test Plan

- Manual testing with a template still in development
- Existing CI for iOS should pass (nothing should be broken)

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by wschurman — COMMENTED** [review]
Looking at the commit history and open PRs on the ASN1Decoder repo, I think it's unlikely a PR will get merged and published, but I still think it's worth a shot (in parallel to this PR).

Just a few inline comments. Will defer to others for the higher-level review about tvos since I've never dealt with it.

File: packages/expo-constants/ios/EXConstantsService.m
Comment (douglowder) [reply packages/expo-constants/ios/EXConstantsService.m:113]: Good catch, will do

File: packages/expo-av/ios/EXAV/EXAV.m
Comment (douglowder) [reply packages/expo-av/ios/EXAV/EXAV.m:880]: Ok -- we can go with `E_UNSUPPORTED_PLATFORM` maybe

**Comment by douglowder** [comment]
> Looking at the commit history and open PRs on the ASN1Decoder repo, I think it's unlikely a PR will get merged and published, but I still think it's worth a shot (in parallel to this PR).

PR has been submitted: https://github.com/filom/ASN1Decoder/pull/39

File: packages/expo-av/ios/EXAV/EXAV.m
Comment (Kudo) [inline packages/expo-av/ios/EXAV/EXAV.m:16]: in expo-av, do you have any better solutions than these if checks?
it would be hard to maintain those checks from time to time, especially we don't have a ci to build tv variants. we will not know if we break anything on tv during new feature development or refactoring.

some idea includes adding an AudioRecordingDelegate to put permission requests, audio recordings, and everything related inside the abstract class. ios and tv could have their own implementations.

Comment (douglowder) [reply packages/expo-av/ios/EXAV/EXAV.m:16]: I'll have a look at this, to see if there is a more maintainable way to support this.

We will have CI for TV once I'm done with the TV changes.

For now, I'll remove expo-av from the list of supported modules, and add that in a separate PR.

**Review by wschurman — APPROVED** [review]
Agree about adding CI for this in a follow-up. Otherwise this will break pretty quickly since most of us ever build for TV locally.

Approving for my addressed comments, but wait for an approval from @Kudo as well to land as he is more familiar with it.

File: packages/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift
Comment (tsapeta) [inline packages/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift:299]:
```suggestion
#if !os(tvOS)
```

File: packages/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift
Comment (tsapeta) [inline packages/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift:232]:
```suggestion
#endif
```

File: packages/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift
Comment (tsapeta) [inline packages/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift:318]:
```suggestion
#endif
```

File: packages/expo-application/CHANGELOG.md
Comment (tsapeta) [inline packages/expo-application/CHANGELOG.md:9]:
```suggestion
- Added support for Apple tvOS ([#24329](https://github.com/expo/expo/pull/24329) by [@douglowder](https://github.com/douglowder))
```

File: packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m
Comment (tsapeta) [inline packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m:1081]: This will have to be backported to Swift if my PR https://github.com/expo/expo/pull/23943 lands first (not likely)

File: packages/expo-updates/ios/EXUpdates/UpdatesUtils.swift
Comment (tsapeta) [inline packages/expo-updates/ios/EXUpdates/UpdatesUtils.swift:33]:
```suggestion
#if os(tvOS)
```

File: packages/expo-updates/ios/EXUpdates/UpdatesUtils.swift
Comment (tsapeta) [inline packages/expo-updates/ios/EXUpdates/UpdatesUtils.swift:35]:
```suggestion
#else
```

File: packages/expo-updates/ios/EXUpdates/UpdatesUtils.swift
Comment (tsapeta) [inline packages/expo-updates/ios/EXUpdates/UpdatesUtils.swift:37]:
```suggestion
#endif
```

File: packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m
Comment (douglowder) [reply packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m:1081]: 👍

**Review by tsapeta — APPROVED** [review]

**Review by Kudo — APPROVED** [review]

**Pull request merged**
